kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/internal/KubernetesDeserializer.java

```diff
@@ -82,7 +82,7 @@ private static KubernetesResource fromObjectNode(JsonParser jp, JsonNode node) t
         if (key != null) {
```

Comment:
I tried these changes with removing FallbackHasMetadata.  Some of the raw logic was written with tests that do not specify the kind in a create.  I'm not sure if that's allowed, but we could easily account for that here by using the generic logic when either the key or the resourceType is null.

Reply:
I cannot think of a case (besides JKube's Fragments) where you can define a resource and not specify the kind and the apiVersion.

We could modify to always try to deserialize to GenericKubernetesResource and let Jackson fail. However I'm not sure of the implications or backwards compatibility side-effects of this change:

```java
    private static KubernetesResource fromObjectNode(JsonParser jp, JsonNode node) throws IOException {
        String key = getKey(node);
        if (key != null) {
            Class<? extends KubernetesResource> resourceType = mapping.getForKey(key);
            if (resourceType != null && KubernetesResource.class.isAssignableFrom(resourceType)){
                return jp.getCodec().treeToValue(node, resourceType);
            }
        }
        return jp.getCodec().treeToValue(node, GenericKubernetesResource.class);
    }
```

Reply:
After trying to modify test-rawcustomresource.yml, I see kind is definitely required.  So CustomResourceCrudTest and a unit test that was added in the mock logic need to be updated to include kind to incorporate your changes.  With that understanding what you had to start with is good.